src/main/java/org/folio/circulation/support/SingleRecordFetcher.java

```diff
@@ -70,4 +71,16 @@ public CompletableFuture<Result<T>> fetch(String id) {
       .thenApply(flatMapResult(interpreter::apply))
       .exceptionally(CommonFailures::failedDueToServerError);
   }
+
+  public CompletableFuture<Result<T>> fetchWithQueryStringParameters(Map<String, String> queryParameters) {
+    if (log.isInfoEnabled()) {
```

Comment:
you can use supplier to avoid additional check

Reply:
@roman-barannyk is this what you had in mind?

```
Supplier<String> logMessageSupplier = () -> "Fetching " + recordType + " with query parameters: " + sanitizeLogParameter(queryParameters.toString());

if (log.isInfoEnabled()) {
  log.info(logMessageSupplier.get());
}
```